hail/src/main/scala/is/hail/expr/ir/GenericLines.scala

```diff
@@ -107,7 +107,7 @@ object GenericLines {
             return
           }
 
-          line.offset = offset
+          line.setOffset(offset)
```

Comment:
I just hoisted the set offset to the top, but the real logic is when we set lineLength.  I'd propose something like:

```
class GenericLine {
  def setLine(newOffset: Int, newLength: Int): Unit = {
    _offset = newOffset
    _lineLength = newLength
    _str = null
  }
```

And then call it (including with the offset) in the code paths below.

You'll probably still need to capture the offset here:

```
  val startOffset = offset
```

because decoding the line will change the offset.